### PR TITLE
Update JSDOM, improve CI infrastructure

### PR DESCRIPTION
**Pull request by armanbilge**
1. Update JSDOM, cache in CI
2. Update Node version used in CI
3. Configure dependabot

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -49,14 +49,17 @@ ThisBuild / crossScalaVersions := Seq("2.11.12", "2.12.15", "2.13.7", "3.0.2")
...
 ThisBuild / githubWorkflowBuildPreamble ++= Seq(
   WorkflowStep.Use(
-    UseRef.Public("actions", "setup-node", "v2.1.2"),
-    name = Some("Setup NodeJS v14 LTS"),
-    params = Map("node-version" -> "14"),
-    cond = Some("matrix.ci == 'ciNode' || matrix.ci == 'ciJSDOMNodeJS'")),
+    UseRef.Public("actions", "setup-node", "v3"),
+    name = Some("Setup NodeJS v16 LTS"),
+    params = Map("node-version" -> "16", "cache" -> "npm"),
+    cond = Some("matrix.ci == 'ciNode' || matrix.ci == 'ciJSDOMNodeJS'")
```
Comment (sjrd) [inline build.sbt:55]:
```suggestion
    cond = Some("matrix.ci == 'ciNode' || matrix.ci == 'ciJSDOMNodeJS'"),
```

File: build.sbt [hunk 2]
```diff
@@ -49,14 +49,17 @@ ThisBuild / crossScalaVersions := Seq("2.11.12", "2.12.15", "2.13.7", "3.0.2")
...
+  ),
   WorkflowStep.Run(
     List("npm install"),
     name = Some("Install jsdom"),
-    cond = Some("matrix.ci == 'ciJSDOMNodeJS'")))
+    cond = Some("matrix.ci == 'ciJSDOMNodeJS'")
```
Comment (sjrd) [inline build.sbt:60]:
```suggestion
    cond = Some("matrix.ci == 'ciJSDOMNodeJS'"),
```

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Comment (sjrd) [inline .github/dependabot.yml:6]: This addition probably deserves a separate commit, at least.

**Review by sjrd — CHANGES_REQUESTED** [review]
Thanks. Would you mind reorganizing the commits a bit, for example as:
* Update Node.js and setup-node
* Update jsdom (possibility merged with the first one)
* Add dependabot config

**Comment by armanbilge** [comment]
Thanks for the review. If you'll merge my PRs, I'm happy to do things your way :)

**Review by sjrd — APPROVED** [review]
Thanks
